### PR TITLE
Implement slicing for singleton immutable sets.

### LINKED ISSUE #23: Slicing is broken for the single-element implementation of ImmutableSet

### PR DESCRIPTION
**Pull request by spigo900**
Resolves #23.

### CODE REVIEW

File: tests/test_immutableset.py [hunk 1]
```diff
@@ -222,6 +222,28 @@ def test_singleton_index(self):
...
+        s = ImmutableSet.of([1])
+
+        self.assertEqual(s, s[0:1])
+        self.assertEqual(s, s[-1:1])
+
+        self.assertEqual(ImmutableSet.empty(), s[1:5])
+        self.assertEqual(ImmutableSet.empty(), s[51:56])
+        self.assertEqual(ImmutableSet.empty(), s[-5:-1])
+        self.assertEqual(ImmutableSet.empty(), s[-1:0])
+        self.assertEqual(ImmutableSet.empty(), s[-2:0])
```
Comment (berquist) [inline tests/test_immutableset.py]: Maybe leave off the zero of this one so we know the default case works?
```suggestion
        self.assertEqual(ImmutableSet.empty(), s[-2:])
```

File: tests/test_immutableset.py [hunk 2]
```diff
@@ -222,6 +222,28 @@ def test_singleton_index(self):
...
+
+        self.assertEqual(ImmutableSet.empty(), s[5:1])
+
+        self.assertEqual(s, s[0:1:1])
+        self.assertEqual(s, s[1:-50:-1])
+        self.assertEqual(s, s[0:1:5])
+        self.assertEqual(s, s[1:-1:-60])
```
Comment (berquist) [inline tests/test_immutableset.py]: Add a case where the ending index isn't specified, like `[0:]` (similar to my above request).

File: tests/test_immutableset.py [hunk 3]
```diff
@@ -222,6 +222,28 @@ def test_singleton_index(self):
         with self.assertRaises(IndexError):
             s[-25]
 
+    def test_singleton_slice(self):
+        s = ImmutableSet.of([1])
```
Comment (berquist) [inline tests/test_immutableset.py]: Using `ImmutableSet.of` is deprecated, use the `immutableset` module-level function instead.

File: tests/test_immutableset.py [hunk 4]
```diff
@@ -222,6 +222,28 @@ def test_singleton_index(self):
...
+
+        self.assertEqual(s, s[0:1])
+        self.assertEqual(s, s[-1:1])
+
+        self.assertEqual(ImmutableSet.empty(), s[1:5])
```
Comment (berquist) [inline tests/test_immutableset.py]: All the `ImmutableSet.empty()` calls should be replaced with `immutableset()`.

**Comment by spigo900** [comment]
@berquist Thanks for the review. I appreciate the comments. I think I've addressed all of them -- let me know if I missed anything.

**Pull request merged**